### PR TITLE
Document unsupported (and deprecated) token types

### LINKED ISSUE #912: Investigate vulnerability to double-entry point tokens
Per TOB-BALV3-8:

> Some proxied tokens have multiple addresses. If a token with multiple
> addresses is used in the Vault, this token can be drained by calling the settle function since the reserves accounting of the Vault assumes a token only has a single address associated with it.

Celo is such a token. We had a reentrancy issue in v2 with double-entry tokens; is there anything we can do beyond documenting it?

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

This adds explicit documentation of unsupported tokens, which we can then point to in the public docs.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [X] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [N/A] Complex code has been commented, including external interfaces
- [N/A] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #912

### CODE REVIEW

File: pkg/interfaces/contracts/vault/VaultTypes.sol [hunk 1]
```diff
@@ -105,6 +105,56 @@ struct PoolRoleAccounts {
...
+//
+// * Double entrypoint (e.g., old Synthetix tokens, now fixed). These could likewise bypass internal accounting by
+//   registering the token under one address, then accessing it through another. This is especially troublesome
+//   in v3, with the introduction of ERC4626 buffers.
+//
+// * Fee on transfer (e.g., PAXG). The Vault issues credits and debits according to given and calculated token amounts,
+//   and settlement assumes that the send/receive transfer functions transfer exactly the given number of tokens.
+//   If this is not the case, transactions will not settle. Unlike with the other types, which are fundamentally
+//   incompatible, it would be possible to design the Vault to handle this - but it was thought this was too much
+//   complexity and gas to support an edge case.
```
Comment (jubeira) [inline pkg/interfaces/contracts/vault/VaultTypes.sol:128]: I think this is solvable with a router that is aware of the tax; it's just that we haven't tried it yet and is not supported in the current ones.

File: pkg/interfaces/contracts/vault/VaultTypes.sol [hunk 2]
```diff
@@ -105,6 +105,56 @@ struct PoolRoleAccounts {
...
+//   to register such tokens will revert with `InvalidTokenDecimals`. Of course, we must also be able to read the token
+//   decimals, so the Vault only supports tokens that implement `IERC20Metadata.decimals`, and return a value less than
+//   or equal to 18.
+//
+// These types of tokens are supported but discouraged, as they don't tend to play well with AMMs generally.
+//
+// * Very low-decimal tokens (e.g., GUSD). The Vault has been extensively tested with 6-decimal tokens (e.g., USDC),
+//   but going much below that may lead to unanticipated effects due to precision loss, especially with smaller trade
+//   values. Also, the Vault enforces a minimum trade size for security. Using a minimum of 1e6 corresponds to 10,000
+//   GUSD, making it impossible to trade small amounts of low-decimal tokens.
```
Comment (jubeira) [inline pkg/interfaces/contracts/vault/VaultTypes.sol:143]: The 1e6 limit corresponds to the scaled amount. If GUSD has 2 decimals, 1 wei is 1e16 when scaled by the decimal scaling factor. It shouldn't ever hit the minimum trade amount.

**Review by jubeira — APPROVED** [review]
Just a few notes, but LGTM otherwise!

File: pkg/interfaces/contracts/vault/VaultTypes.sol [hunk 3]
```diff
@@ -105,6 +105,55 @@ struct PoolRoleAccounts {
...
+//
+// * Rebasing tokens (e.g., aDAI). The Vault keeps track of token balances in its internal accounting; any token whose
+//   balance changes asynchronously (i.e., outside a swap or liquidity operation), would get out-of-sync with this
+//   internal accounting. This category would also include "airdrop" tokens, whose balances can change unexpectedly.
+//
+// * Double entrypoint (e.g., old Synthetix tokens, now fixed). These could likewise bypass internal accounting by
+//   registering the token under one address, then accessing it through another. This is especially troublesome
+//   in v3, with the introduction of ERC4626 buffers.
+//
+// * Fee on transfer (e.g., PAXG). The Vault issues credits and debits according to given and calculated token amounts,
```
Comment (joaobrunoah) [inline pkg/interfaces/contracts/vault/VaultTypes.sol:124]:
```suggestion
// * Fee on transfer (e.g., PAXG). The Vault issues credits and debts according to given and calculated token amounts,
```

File: pkg/interfaces/contracts/vault/VaultTypes.sol [hunk 4]
```diff
@@ -105,6 +105,55 @@ struct PoolRoleAccounts {
...
+//   and settlement assumes that the send/receive transfer functions transfer exactly the given number of tokens.
+//   If this is not the case, transactions will not settle. Unlike with the other types, which are fundamentally
+//   incompatible, it would be possible to design a Router to handle this - but we didn't try it. In any case, it's
+//   not supported in the current Routers.
+//
+// * Tokens with greater than 18 decimals (e.g., YAM-V2). The Vault handles token scaling: i.e., handling I/O for
```
Comment (joaobrunoah) [inline pkg/interfaces/contracts/vault/VaultTypes.sol:130]:
```suggestion
// * Tokens with more than 18 decimals (e.g., YAM-V2). The Vault handles token scaling: i.e., handling I/O for
```

**Review by joaobrunoah — APPROVED** [review]
LGTM! Just some suggestions that I'm not even sure are correct

File: pkg/interfaces/contracts/vault/VaultTypes.sol (hunk 3 above)
Comment (EndymionJkb) [reply pkg/interfaces/contracts/vault/VaultTypes.sol:124]: This is actually correct (and wrong elsewhere). A "debit" is an "entry in a ledger", which is what we're doing here, vs. the "debt" itself. It's entering credits and debits into the accounting ledger, which should all balance out. credit/debit, asset/(liability or debt)

**Pull request merged**